core/src/main/java/io/jenkins/pluginhealth/scoring/probes/AbstractDeprecatedAnnotationImportsCheckProbe.java

```diff
@@ -0,0 +1,83 @@
+package io.jenkins.pluginhealth.scoring.probes;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import io.jenkins.pluginhealth.scoring.model.Plugin;
+import io.jenkins.pluginhealth.scoring.model.ProbeResult;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.util.CollectionUtils;
+
+/**
+ * The abstract probe checks for deprecated imports in a plugin.
+ */
+public abstract class AbstractDeprecatedAnnotationImportsCheckProbe extends Probe {
+    public static final int ORDER = SCMLinkValidationProbe.ORDER;
```

Comment:
Is it supposed to run at the same time as `SCMLinkValidationProbe` or after it? I believe we need the source code to be available before checking it, so we might need to run after this probe?

Reply:
In the first line: ` context.getScmRepository();` I am fetching the repository link. I want the link to be validated before I fetch the repo. 

So the `SCMLinkValidationProbe` should be executed first. I have also added it in the `getProbeResultRequirement`

Reply:
Sorry, what I meant is: is this OK to use the same value for `ORDER` or should we increment it?

Reply:
https://javatechonline.medium.com/order-annotation-in-spring-boot-499d76c5a29c

In this article, it says:

> If Component does not have @Order annotation, it will execute in alphabetical order of bean name by default. Furthermore, if two components are having the same order value, the execution will happen in the alphabetical order of component name.

You are right. I added + 100 in that case. So that this probe run after `SCMLinkValidationProbe`.